mne/io/nsx/nsx.py

```diff
@@ -0,0 +1,601 @@
+            "setting values to 0 and Nyquist."
+        )
+        info["highpass"] = 0.0
+        info["lowpass"] = info["sfreq"] / 2.0
+
+    # Some keys to be consistent with FIF measurement info
+    info["description"] = None
+
+    info._unlocked = False
+    info._update_redundant()
+
+    orig_format = ORIG_FORMAT
+
+    raw_extras = {
+        key: [r[key] for r in nsx_info["data_header"]]
+        for key in nsx_info["data_header"][0]
+    }
+    for key in raw_extras:
+        raw_extras[key] = np.array(raw_extras[key], int)
+    good_data_packets = raw_extras.pop("header") == 1
+    if not good_data_packets.any():
+        raise RuntimeError("NSx file appears to be broken")
+    raw_extras = {key: raw_extras[key][good_data_packets] for key in raw_extras.keys()}
+    raw_extras["timestamp"] = raw_extras["timestamp"] // nsx_info["period"]
+    first_samp = raw_extras["timestamp"][0]
+    last_samp = raw_extras["timestamp"][-1] + raw_extras["nb_data_points"][-1]
+    n_samples = last_samp - first_samp
+
+    return (
+        info,
+        fname,
+        nsx_info["spec"],
+        n_samples,
+        orig_format,
+        raw_extras,
+        orig_units,
+    )
+
+
+def _check_stim_channel(stim_channel, ch_names):
+    """Check that the stimulus channel exists in the current datafile."""
+    DEFAULT_STIM_CH_NAMES = ["status", "trigger"]
+
+    if stim_channel is None or stim_channel is False:
+        return [], []
+
+    if stim_channel is True:  # convenient aliases
+        stim_channel = "auto"
+
+    if isinstance(stim_channel, str):
+        if stim_channel == "auto":
+            if "auto" in ch_names:
+                warn(
+                    RuntimeWarning,
+                    "Using `stim_channel='auto'` when auto"
+                    " also corresponds to a channel name is ambiguous."
+                    " Please use `stim_channel=['auto']`.",
+                )
+            else:
+                valid_stim_ch_names = DEFAULT_STIM_CH_NAMES
+        else:
+            valid_stim_ch_names = [stim_channel.lower()]
+
+    elif isinstance(stim_channel, int):
+        valid_stim_ch_names = [ch_names[stim_channel].lower()]
+
+    elif isinstance(stim_channel, list):
+        if all([isinstance(s, str) for s in stim_channel]):
+            valid_stim_ch_names = [s.lower() for s in stim_channel]
+        elif all([isinstance(s, int) for s in stim_channel]):
+            valid_stim_ch_names = [ch_names[s].lower() for s in stim_channel]
+        else:
+            raise ValueError("Invalid stim_channel")
+    else:
+        raise ValueError("Invalid stim_channel")
+
+    ch_names_low = [ch.lower() for ch in ch_names]
+    found = list(set(valid_stim_ch_names) & set(ch_names_low))
+
+    if not found:
+        return [], []
+    else:
+        stim_channel_idxs = [ch_names_low.index(f) for f in found]
+        names = [ch_names[idx] for idx in stim_channel_idxs]
+        return stim_channel_idxs, names
```

Comment:
I don't think the conditional is necessary -- the following code should result in two empty lists -- so let's not add it (the iteration over an empty list will be near instantaneous, and every branch is more things to change/test/think about):
```suggestion
    stim_channel_idxs = [ch_names_low.index(f) for f in found]
    names = [ch_names[idx] for idx in stim_channel_idxs]
    return stim_channel_idxs, names
```